Sources/PackageModel/Module.swift

```diff
@@ -119,19 +122,31 @@ extension ClangModule: XcodeModuleProtocol {
     }
 }
 
-public class TestModule: SwiftModule {
+let testModuleNameSuffix = "TestSuite"
+
+public protocol TestModuleProtocol {
     /// The base prefix for the test module, used to associate with the target it tests.
-    //
-    // FIXME: This seems like a fragile way to model this relationship.
-    public let basename: String
-    
-    public init(basename: String, sources: Sources) throws {
-        self.basename = basename
-        try super.init(name: basename + "TestSuite", sources: sources)
-        c99name = try PackageModel.c99name(name: basename) + "TestSuite"
+    var basename: String { get }
+    /// The name of the module.
+    var name: String { get }
+
+    associatedtype T
```

Comment:
What is this associated type?

It seems a little weird to have this protocol while still having the isTest property on the module, should this all just get folded in?